data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/sink/FileSink.java

```diff
@@ -56,18 +57,31 @@ public FileSink(final String outputFile) {
     }
 
     @Override
-    public void output(Collection<Record<String>> records) {
+    public void output(Collection<Record<Object>> records) {
         try (BufferedWriter writer = Files.newBufferedWriter(Paths.get(outputFilePath),
                 StandardCharsets.UTF_8)) {
-            for (final Record<String> record : records) {
-                writer.write(record.getData());
-                writer.newLine();
+            for (final Record<Object> record : records) {
+                checkTypeAndWriteObject(record.getData(), writer);
             }
         } catch (IOException ex) {
             throw new RuntimeException(format("Encountered exception opening/creating file %s", outputFilePath), ex);
         }
     }
 
+    // Temporary function to support both trace and log ingestion pipelines.
+    // TODO: This function should be removed with the completion of: https://github.com/opensearch-project/data-prepper/issues/546
+    private void checkTypeAndWriteObject(final Object object, final BufferedWriter writer) throws IOException {
+        if (object instanceof String) {
+            writer.write((String) object);
+            writer.newLine();
+        } else if (object instanceof Event) {
+            writer.write(((Event) object).toJsonString());
+            writer.newLine();
+        } else {
+            throw new RuntimeException("Invalid record type. FileSink only supports String and Events");
```

Comment:
Is the best response to raise an exception here?

1. We could still publish the object after converting it to a string. The result may not be great but would alert the customer something is not working as expected.
2. Throwing the exception here would prevent any other (potentially good) records in the list from being written to the sink.

Reply:
@sbayer55 Good point. L81 is supposed to be transient as eventually we will only support Event type, i.e. change from Sink<Record<Object>> into Sink<Record<Event>>. Despite that, logging the error will be a better option than throwing a RuntimeException.

> We could still publish the object after converting it to a string.
The idea of specifying Record type is to unify the format. So we should avoid publishing malformatted object.